docs/content/configuration/auth.md

```diff
@@ -29,12 +29,36 @@ A request will pass through all Authenticators in the chain, until one of the Au
 
 If no Authenticator in the chain successfully authenticated a request or sent an HTTP error response, an HTTP error response will be sent at the end of the chain.
 
-Druid includes a built-in Authenticator, used for the default unsecured configuration.
+Druid includes two built-in Authenticators, one of which is used for the default unsecured configuration.
 
 ### AllowAll Authenticator
 
 This built-in Authenticator authenticates all requests, and always directs them to an Authorizer named "allowAll". It is not intended to be used for anything other than the default unsecured configuration.
 
+### Anonymous Authenticator
+
+This built-in Authenticator authenticates all requests, and directs them to an Authorizer specified in the configuration by the user. It is intended to be used for adding a default level of access.
+The Anonymous Authenticator would be added to the end of the authentication chain so if a user sends a request which fails the first part of the chain, the request will succeed or fail depending on how the Authorizer linked to the Anonymous Authenticator is configured.
```

Comment:
I would clarify this a bit:

>if a user sends a request which fails the first part of the chain

From the `getFilter()` method docs in `Authenticator`:

```
```
   * If a filter cannot recognize a request's format (e.g., the request does not have credentials compatible
   * with a filter's authentication scheme), the filter should not send an error response, allowing other
   * filters to handle the request. A challenge response will be sent if the filter chain is exhausted.
   * <p>
   * If the authentication fails (i.e., a filter recognized the authentication scheme of a request, but the credentials
   * failed to authenticate successfully) the Filter should send an error response, without needing to proceed to
   * other filters in the chain..
```
```

So a request would reach the anonymous authenticator at the end of the chain if the request didn't have any credentials set, but if the request attempted to authenticate against some scheme (e.g., Basic HTTP by including a `Authorization: Basic <credentials>` header), then the request would immediately terminate at the Authenticator which recognized the request's credentials.

Reply:
I agree that the docs should be updated in line with what you commented. However, I came across the following situation. With `druid.auth.authenticatorChain=["MyBasicAuthenticator", "anonymous"]` set, requests that should have failed basic authentication (e.g incorrect username/password) seemed to be passed onto the next filter in the chain.

A response of 403 was returned (because the permissions for the default user in the anonymous authenticator did not authorize the user to access that particular resource) instead of the expected 401. Then, it is possible to update the permissions of the default user such that the same request succeeds. So it seems like the second bullet point above is incorrect.

Are you able to reproduce this issue?

Reply:
Ah, it looks like BasicHTTPAuthenticator needs to be updated to adhere to that method contract, thanks for reporting that. I'll make a PR for that fix (calling `        httpResp.sendError(HttpServletResponse.SC_UNAUTHORIZED)` if the `checkCredentials` call fails)

Can you make the doc change here? I'll merge this PR after that